sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -103,7 +103,15 @@ function appendPath(url: string, pathToAppend?: string): string {
     pathToAppend = pathToAppend.substring(1);
   }
 
-  newPath = newPath + pathToAppend;
+  const searchStart = pathToAppend.indexOf("?");
+  if (searchStart !== -1) {
+    const path = pathToAppend.substring(0, searchStart);
+    const search = pathToAppend.substring(searchStart + 1);
+    newPath = newPath + path;
+    parsedUrl.search = parsedUrl.search ? `${parsedUrl.search}&${search}` : search;
```

Comment:
Do we need to make sure that search is non-empty before appending it with `&`? It'd be an extremely unlikely case that you'd have `pathToAppend` end with `?` but not sure what would happen the query part ended with an `&`

Reply:
I can add a check